imap/backend.c

```diff
@@ -1393,3 +1393,12 @@ EXPORTED int backend_version(struct backend *be)
                         __func__, be->banner);
     return 6;
 }
+
+EXPORTED int backend_supports_sieve_mailbox(struct backend *be)
+{
+    if (strstr(be->banner, "Cyrus IMAP 3.5")) {
+        return 1;
+    }
+
+    return 0;
+}
```

Comment:
Ouch, I don't like this -- this then becomes another function like `backend_version()`, where we have to remember to update it on every branch every time we make a new release.

It's also awkward because "3.5" only supports sieve-in-mailbox if it's a new enough 3.5 to have this PR merged in, but there's a whole years worth of "3.5" builds that didn't.  (Though this is more of a theoretical problem than a real one, since nobody but us uses 3.5 yet, and we don't use XFER.)

I think sieve-in-mailbox warrants bumping MAILBOX_MINOR_VERSION to 18 (and in hindsight I think we probably ought to have bumped it for uuid-mailboxes too, but that's sailed already).

That being the case, this function could then be something like this (or even just a macro):

```
EXPORTED int backend_supports_sieve_mailbox(struct backend *be)
{
    /* sieve mailbox was introduced with MAILBOX_MINOR_VERSION 18 */
    return backend_version(be) >= 18;
}
```

And then the only time we'd ever need to update it again is if some far future version moves sieve storage somewhere else.

I don't remember all the ramifications of bumping MAILBOX_MINOR_VERSION; @brong does this idea seem sane?